expression/builtin_encryption_test.go

```diff
@@ -227,7 +227,14 @@ func TestAESDecrypt(t *testing.T) {
 		for _, param := range tt.params {
 			args = append(args, types.NewDatum(param))
 		}
-		f, err := fc.getFunction(ctx, datumsToConstants(args))
+		// Set charset and collate for args to test new charset support
+		cons := datumsToConstants(args)
+		chs, col := ctx.GetSessionVars().GetCharsetInfo()
+		// Start from 1, as first arg is crypt_str
+		for i := 1; i < len(cons); i++ {
+			cons[i].GetType().Charset, cons[i].GetType().Collate = chs, col
+		}
```

Comment:
I think L226 - L236 could change to 
```
args := datumsToConstants([]types.Datum{fromHex(tt.crypt)})
args = append(args, primitiveValsToConstants(ctx, tt.params)...)
```

The `primitiveValsToConstants` func is similar as your code here.

Reply:
Resolved.